WiseUnpacker/SabreTools.Serialization.Deserializers/WiseSectionHeader.cs

```diff
@@ -87,39 +87,193 @@ public class WiseSectionHeader : BaseBinaryDeserializer<SectionHeader>
+                if (headerLength != 8)
+                {
+                    header.ThirdExecutableFileEntryLength = data.ReadUInt32LittleEndian();
+                    header.UnknownValue10 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue11 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue12 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue13 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue14 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue15 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue16 = data.ReadUInt32LittleEndian();
+                    header.UnknownValue17 = data.ReadUInt32LittleEndian();
+                    if (headerLength != 17)
+                    {
+                        header.UnknownValue18 = data.ReadUInt32LittleEndian();
+                    }
+                }
+            }
+            
 
-            header.UnknownValue7 = data.ReadUInt32LittleEndian();
-            header.UnknownValue8 = data.ReadUInt32LittleEndian();
-            if (headerLength == 8)
-                return header;
+            // Parse strings
+            // TODO: Count size of string section for later size verification
+            
+            data.Seek(initialOffset + localWisOffset, 0);
+            header.TmpString = data.ReadNullTerminatedString(Encoding.ASCII); // read .TMP string
+            header.GuidString = data.ReadNullTerminatedString(Encoding.ASCII); // read GUID string
+            // TODO: better way to figure out how far it's needed to advance?
+            if (header.Version == null)
+            {
+                // TODO: error output
+                return null;
+            }
+            int versionSize;
+            if (header.Version[header.Version.Length - 1] == 0x02)
+            {
+                versionSize = header.Version[header.Version.Length - 3];
+            }
+            else
+            {
+                versionSize = header.Version[header.Version.Length - 2];
+            }
+            if (versionSize != 1) // third byte seems to indicate size of NonWiseVer
+            {
 
-            header.UnknownValue9 = data.ReadUInt32LittleEndian();
-            header.UnknownValue10 = data.ReadUInt32LittleEndian();
-            header.UnknownValue11 = data.ReadUInt32LittleEndian();
-            header.UnknownValue12 = data.ReadUInt32LittleEndian();
-            header.UnknownValue13 = data.ReadUInt32LittleEndian();
-            header.UnknownValue14 = data.ReadUInt32LittleEndian();
-            header.UnknownValue15 = data.ReadUInt32LittleEndian();
-            header.UnknownValue16 = data.ReadUInt32LittleEndian();
-            header.UnknownValue17 = data.ReadUInt32LittleEndian();
-            if (headerLength == 17)
-                return header;
+                byte[] stringBytes = data.ReadBytes(versionSize);
```

Comment:
Consider splitting some of this string reading stuff into a helper method. I don't have any concrete suggestions as to how, but I just noticed how long this single method is.

Reply:
I've split it into two helper functions.